internal/mode/static/state/graph/route_common.go

```diff
@@ -529,6 +534,29 @@ func routeAllowedByListener(
 	return true
 }
 
+// isRouteKindAllowedByListener checks if the route is allowed to attach to the listener.
+// If the listener specifies allowed kinds, the route kind must be in the list.
+// If the listener does not specify allowedRoutes, allowed routes are determined using the listener protocol.
+func isRouteTypeAllowedByListener(listener *Listener, routeType RouteType) bool {
```

Comment:
Can we take advantage of the[ Listener.SupportedKinds](https://github.com/nginxinc/nginx-gateway-fabric/blob/01c46ae841bc51210eb97208da7b483e669eaf79/internal/mode/static/state/graph/gateway_listener.go#L35) field?

When we build the listener, we set the SupportedKinds [here](https://github.com/nginxinc/nginx-gateway-fabric/blob/main/internal/mode/static/state/graph/gateway_listener.go#L227). 

We should be able to do something like:

```
routeKind := convert(routeType)
for _, kind := range listener.SupportedKinds {
    if kind.Kind == routeKind {
          return true
    }
}

return false
```

Reply:
The supported kinds [function](https://github.com/nginxinc/nginx-gateway-fabric/blob/main/internal/mode/static/state/graph/gateway_listener.go#L227) will need a little tweaking so that we are actually setting the supported kinds based on the protocol when allowed kinds isn't set. But I think it would be better if all of the logic around parsing allowed routes is in one place.